src/js/components/chart/utils-normalization.js

```diff
@@ -0,0 +1,219 @@
+// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP
+
+import zip from 'lodash/zip';
+import flatten from 'lodash/flatten';
+import isInteger from 'lodash/isInteger';
+import sortedIndexOf from 'lodash/sortedIndexOf';
+
+
+/**
+ * The normalization class includes a normalize method in order to help
+ * the Chart consumers to normalize the series.
+ *
+ * Chart component assumes the array of values
+ * (whether its for Line/Bar/Area/Axis) is always normalize and the points are
+ * evenly spreader on the array.
+ *
+ * normalize func will normalize values in case of data-gaps and blank areas,
+ * and will fill 'undefined' for the missing points added during normalization
+ */
+
+export class normalization {
+
+  constructor() {
+    this.getXAxis = this.getXAxis.bind(this);
+    this.getXValues = this.getXValues.bind(this);
+    this.getMinVal = this.getMinVal.bind(this);
+    this.getMinVal = this.getMinVal.bind(this);
+  }
+
+
+  /**
+   * Getting the xAxis values form a given array data structure
+   *
+   * @param data
+   * return an array of all the xValues from the series map
+   */
+  getXValues (data) {
+    if (!data) return [];
+    return data.map(dataPoint => parseInt(dataPoint[0], 10));
+  };
+
+  /**
+   * With a given a array of 2D arrays, the function will return the common
+   * xAxis for all data-points distribute evenly
+   *
+   * Example 1 (granularity = 1):
+   * series = [
+   *           [[1,2],[3,3],[4,5]], //series
+   *           [[0,4],[6,7]]
+   *          ]
+   *
+   * return xAxis => [0,1,2,3,4,5,6]
+   *
+   *
+   * Example 2 (granularity = 2):
+   * series = [
+   *           [[2,2],[4,3],[6,5]], //series
+   *           [[4,4],[8,7]]
+   *          ]
+   *
+   * return xAxis => [2,4,6,8]
+   *
+   *
+   * Example 3 (granularity = 2):
+   * series = [
+   *           [[2,2],[6,5]],
+   *           [[4,4],[10,7]]
+   *          ]
+   *
+   * return xAxis => [2,4,6,8,10]
+   *
+   *
+   * @param series array of 2D arrays
+   * @param granularity the interval of the data points, uses '1' as default
+   */
+
+  getXAxis(series, granularity) {
+    let flat = series;
+    // Check whether the series has more than one set of values
+    if (Array.isArray(series[0][0])) {
+      flat = flatten(series);
+    }
+
+    const xValues = this.getXValues(flat);
+    const max = this.getMaxVal(xValues);
+    const min = this.getMinVal(xValues);
+
+    if (!granularity) {
+      // Possible to implement a calculator that with a given data points
+      // will calculate the optimized granularity
+      granularity = 1;
+      console.warn('granularity was not defined for getXAxis calculation');
```

Comment:
If the user will use timeseries (more than 8 digits number) for the data, and neglect to send granularity, all of the UI will be stack due to the complexity of the calculation. this is why I thought it will be nice to indicate it. 
In HF for eample, we don't use granularity = 1 in any of our analytics